### PR TITLE
handling POST/ rooms route

### PR DESCRIPTION
**Pull request by LinaYahya**
Relates #19 
- add query to add room
- validation for input name done
- controller add room created ( verifying user admin or not then add room by query)

### CODE REVIEW

File: server/src/app.js [hunk 1]
```diff
@@ -26,8 +26,6 @@ if (process.env.NODE_ENV === 'development') app.use(morgan('dev'));
 
 app.use('/api/v1/', router);
 
-if (process.env.NODE_ENV === 'development') app.use(morgan('dev'));
```
Comment (Amoodaa) [inline server/src/app.js:29]: Why did you remove morgan

File: server/src/controllers/room.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    .then(() => verifyCookie(req.cookies.token))
+    .then(({ role }) => {
+      if (role) {
+        return getRoom(name);
+      }
+      throw Boom.unauthorized('This action just for admin');
+    })
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        addNewRoom(name);
```
Comment (Amoodaa) [inline server/src/controllers/room.js:23]: I think you need to return the promise

File: server/src/controllers/room.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const { addNewRoom, getRoom } = require('../database/queries');
+
+const addRoom = (req, res, next) => {
+  const { name } = req.body;
+  roomSchema
+    .validateAsync({ name })
+    .catch((error) => {
+      throw Boom.badRequest(error.message);
+    })
+    .then(() => verifyCookie(req.cookies.token))
```
Comment (Amoodaa) [inline server/src/controllers/room.js:14]: This should be as a middleware, that puts the cookies content in req.user
and you can access user_id and role like req.user.role

Comment (Amoodaa) [reply server/src/controllers/room.js:14]: this makes this pull request pending to omar's #45

File: server/src/app.js (hunk 1 above)
Comment (LinaYahya) [reply server/src/app.js:29]: cause it written twice, i just removed the repeated line

File: server/src/controllers/room.js (hunk 2 above)
Comment (LinaYahya) [reply server/src/controllers/room.js:23]: nothing return from INSERT query so the next promise would be for undefined and that's cause an error

File: server/__test__/database.test.js [hunk 4]
```diff
@@ -14,9 +19,23 @@ test('testing checkemail query so it expect to return the row from userbooking b
...
   }));
 
+test('test get room query to get the row by given name', () =>
+  getRoom('Tokyo').then((result) => {
+    const { name } = result.rows[0];
+    expect(name).toEqual('Tokyo');
+  }));
+
+test('add room query', () => {
+  addNewRoom('Cairo').then((result) => {
```
Comment (Amoodaa) [inline server/__test__/database.test.js:29]: return the promise so jest can know this is asynchronous code

File: server/__test__/database.test.js [hunk 5]
```diff
@@ -14,9 +19,23 @@ test('testing checkemail query so it expect to return the row from userbooking b
...
+    const { name } = result.rows[5];
+    expect(name).toEqual('Cairo');
+  });
+});
+
 test('deleteUserById query', () => {
   deleteUser('3').then(() => {
```
Comment (Amoodaa) [inline server/__test__/database.test.js:36]: and return the promsie here, fix this test on your way please xd

File: server/src/controllers/room.js (hunk 2 above)
Comment (Amoodaa) [reply server/src/controllers/room.js:16]: if you dont return a promise, the promise resolves immediately to undefined, as a result the response would be 201, and the data did not actually change

File: server/src/database/queries/addRoom.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+const connection = require('../config/connection');
+
+const addNewRoom = (name) =>
+  connection.query('INSERT INTO room VAlUES = $1', [name]);
```
Comment (Amoodaa) [inline server/src/database/queries/addRoom.js:4]: query syntax is wrong
